### PR TITLE
enable downloads on lbry.tv

### PR DESCRIPTION
**Pull request by seanyesmunt**
#### Todo
- [x] Trigger view event for download click
- [x] ~Fix html/pdf viewer~ api side fix needed

### CODE REVIEW

File: ui/component/fileDownloadLink/view.jsx [hunk 1]
```diff
@@ -15,10 +17,42 @@ type Props = {
...
+  function handleDownload() {
+    // @if TARGET='app'
+    download(uri);
+    // @endif;
+    // @if TARGET='web'
+    triggerViewEvent(uri);
+    // @endif;
+  }
+
+  if (IS_WEB && isPaidContent) {
```
Comment (kauffj) [inline ui/component/fileDownloadLink/view.jsx]: When do we `IS_WEB` and when do we `@if TARGET='web'`?

File: ui/component/fileDownloadLink/view.jsx [hunk 2]
```diff
@@ -41,19 +75,21 @@ function FileDownloadLink(props: Props) {
...
+
+  return (
+    <ToolTip label={IS_WEB ? __('Download') : __('Add to your library')}>
+      <Button
+        button="alt"
+        icon={ICONS.DOWNLOAD}
+        onClick={handleDownload}
+        // @if TARGET='web'
+        download={fileName}
+        href={downloadUrl}
```
Comment (kauffj) [inline ui/component/fileDownloadLink/view.jsx]: I'd like `downloadUrl` to only get set on lbry.tv

Comment (kauffj) [reply ui/component/fileDownloadLink/view.jsx]: (Yes, I understand it isn't used)

**Comment by kauffj** [comment]
@seanyesmunt nm, I was wrong in my other review comment. No blocking feedback.

File: ui/util/lbrytv.js [hunk 3]
```diff
@@ -9,5 +9,10 @@ function generateEmbedUrl(claimName, claimId) {
   return `${URL}/$/embed/${claimName}/${claimId}`;
 }
 
+function generateDownloadUrl(claimName, claimId, apiUrl) {
+  const streamUrl = generateStreamUrl(claimName, claimId, apiUrl);
```
Comment (kauffj) [inline ui/util/lbrytv.js]: Can these also get wrapped similar to embed links for posterity/readability? E.g. lbry.tv/$/download/${claimName}/${claimId}

Comment (seanyesmunt) [reply ui/util/lbrytv.js]: Yeah. Good idea

**Pull request merged**